### PR TITLE
executor: change struct `row` to []types.Datum

### PR DESCRIPTION
**Pull request by winoros**
Simplify the struct since `Row` only has a member `Data []types.Datum`.
PTAL @shenli @hanfei1991 @zimulala @tiancaiamao

### CODE REVIEW

File: executor/union_scan.go [hunk 1]
```diff
@@ -125,19 +125,40 @@ func (us *UnionScanExec) Next() (*Row, error) {
 		if row == nil {
 			return nil, nil
 		}
-		if row == snapshotRow {
```
Comment (hanfei1991) [inline executor/union_scan.go]: Can't we compare the two pointers?

File: executor/write.go [hunk 2]
```diff
@@ -1165,14 +1165,14 @@ type UpdateExec struct {
 	updatedRowKeys map[int64]map[int64]struct{}
 	tblID2table    map[int64]table.Table
 
-	rows        []*Row          // The rows fetched from TableExec.
+	rows        []Row           // The rows fetched from TableExec.
```
Comment (hanfei1991) [inline executor/write.go]: It seems there is a redundant space char.

Comment (winoros) [inline executor/write.go]: This space is to align the comment at the right side...

**Comment by ngaut** [comment]
What's the performance impact ?

File: executor/union_scan.go (hunk 1 above)
Comment (winoros) [inline executor/union_scan.go]: Code like `&row == &snapshotRow`? If so, we cannot do that.

**Comment by zimulala** [comment]
LGTM

**Comment by winoros** [comment]
@ngaut compare with current master, test by sysbench `select.sh`:
original total query is `14569737 (24282.38 per sec.)`
current branch is `15118160 (25196.77 per sec.)`

**Pull request merged**